src/storage/src/table/batch_table/storage_table.rs

```diff
@@ -626,6 +626,86 @@ impl<S: StateStore, SD: ValueRowSerde> StorageTableInner<S, SD> {
         .await
     }
 
+    /// Iterates on the table with the given prefix of the pk in `pk_prefix` and the range bounds.
+    /// Returns a stream of chunks of columns with the provided `chunk_size`
+    async fn chunk_iter_with_pk_bounds(
+        &self,
+        epoch: HummockReadEpoch,
+        pk_prefix: impl Row,
+        range_bounds: impl RangeBounds<OwnedRow>,
+        ordered: bool,
+        chunk_size: usize,
+        prefetch_options: PrefetchOptions,
+    ) -> StorageResult<impl Stream<Item = StorageResult<(Vec<ArrayRef>, usize)>> + Send> {
+        use risingwave_common::util::iter_util::ZipEqFast;
+
+        let iter = self
+            .iter_with_pk_bounds(epoch, pk_prefix, range_bounds, ordered, prefetch_options)
+            .await?;
+
+        // Uses ArrayBuilderImpl instead of DataChunkBuilder here to demonstrate how to build chunk in a columnar manner
```

Comment:
I think we can have a separate method that convert the iter to the stream via the `try_stream` macro. It can make the the state management more elegant. The code can be as simple as
```
#[try_stream(ok = Vec<ArrayRef>, error = StorageError)]
async fn convert_iter_to_array_vec_stream(mut iter) {
    let mut builder = new_builder();
    let mut row_count = 0;
    while let Some(row) = iter.try_next().await? {
        builder.append(row);
        row_count += 1;
        if row_count == chunk_size {
            yield builder.build();
            row_count = 0;
            builder = new_builder();
        }
    }
    if row_count > 0 {
        yield builder.build();
    }
}
```